centaur/src/main/scala/centaur/test/ObjectCounter.scala

```diff
@@ -38,6 +39,22 @@ object ObjectCounterInstances {
       storage.list(g.bucket, BlobListOption.prefix(g.directory)).iterateAll.asScala
     listObjectsAtPath(_).size
   }
+
+  implicit val blobObjectCounter: ObjectCounter[BlobContainerClient] = (containerClient : BlobContainerClient) => {
+    val pathToInt: Path => Int = providedPath => {
+      //Our path parsing is somewhat GCP centric. Convert to a blob path starting from the container root.
+      def pathToBlobPath(parsedPath : Path) : String =  {
+        if(parsedPath.bucket.isEmpty) return ""
+        if(parsedPath.directory.isEmpty) return parsedPath.bucket
+        return parsedPath.bucket + "/" + parsedPath.directory
```

Comment:
This seems like a great place for pattern matching. There may be a few different ways to do this but roughly I think this may be a more scala-y way to approach this

```suggestion
       return (Option(parsedPath.bucket), Option(parsedPath.directory)) match 
          case (None, _ )  => ""
          case (Some(bucket), None) => bucket
          case (Some(bucket), Some(directory)) => bucket + "/" + directory
```

Reply:
Thanks! Will try it out.

Reply:
Do you really mean to use `return` here? The use of `return` is [generally discouraged](https://www.baeldung.com/scala/return-keyword) in Scala and was generally avoided in the past.

Reply:
Nope, that was an old habit, thanks for calling this out!